src/runtime/mod.rs

```diff
@@ -130,14 +115,36 @@ impl Runtime {
 
 impl Drop for Runtime {
     fn drop(&mut self) {
-        // drop tasks
+        // drop tasks in correct order
         unsafe {
             ManuallyDrop::drop(&mut self.local);
-            ManuallyDrop::drop(&mut self.rt);
+            ManuallyDrop::drop(&mut self.tokio_rt);
         }
     }
 }
 
+fn start_uring_wakes_task(
+    tokio_rt: &tokio::runtime::Runtime,
+    local: &LocalSet,
+    driver: driver::Handle,
+) {
+    let _guard = tokio_rt.enter();
+    let async_driver_handle = AsyncFd::new(driver).unwrap();
+
+    local.spawn_local(drive_uring_wakes(async_driver_handle));
+}
+
+async fn drive_uring_wakes(driver: AsyncFd<driver::Handle>) {
```

Comment:
`drive_uring_wakes` is not bad. By which I mean, I kind of like it. It's very descriptive. Unfortunately it doesn't call a `uring_wakes`. Maybe `drive_uring_sync` or `drive_cq_sync`?

And would you mind using whatever name you pick for the argument on line 134, as the parameter name here? Or pick a super short name, like the Go developers and old programmers like, since the type is everything and the name not so much. `afd` for example.
I just can't bring myself to thinking of it as a driver too.

In case I seem annoying, first I apologize. Second, look at line 140. The driver is not readable, the async_fd is readable, or becomes readable once the Future is resolved. True the unwrap yields a guard to the driver's Handle. I can think of it as a guard handle, even if the name is just guard.

Reply:
I think it's probably best for us to describe what we are doing here, which isn't just regarding syncing the cqueue, but more importantly waking tasks.